### PR TITLE
first two tasks

### PR DESCRIPTION
**Pull request by NoxVail**

### CODE REVIEW

File: Tasks/ruby_objects/2.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  def self.define_splash_screen
+    a = ObjectSpace.each_object(MarioGame).to_a
+    b = a.sort_by { |x| x.level_number}.last
+    b.define_singleton_method(:splash) do
+      'Splash Screen'
+    end
+  end
+end
+
+mar = binding
```
Comment (Patriarh22) [inline Tasks/ruby_objects/2.rb]: You can use `instance_variable_set` and `instance_variable_get` methods without `binding`, you don't need it

File: Tasks/ruby_objects/1-1.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require_relative 'PrintMachine'
+
+class MarioGame
+  include PrintMachine
```
Comment (Patriarh22) [inline Tasks/ruby_objects/1-1.rb]: Add one more line between `include` and `initialize` method

File: Tasks/ruby_objects/2.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  def show_count_of_enemies
+    "Count of enemies on level is: #{@enemies}"
+  end
+end
+
+class MarioGame
+  include PrintMachine
+
+  attr_accessor :level_background, :count_of_enemies, :speed_enemies, :level_number
```
[the remark this answers is not in the document]
Comment (Patriarh22) [reply Tasks/ruby_objects/2.rb]: Remove please that `attr_accessor`s

File: Tasks/ruby_objects/1-2.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  def change_count_of_enemies(new_count_of_enemies)
+    @enemies = new_count_of_enemies
+  end
+
+end
+
+class MyGame
+  include PrintMachine
+
+  def initialize(bg, ln, se, en, jp)
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/1-2.rb]: Done

File: Tasks/ruby_objects/1-2.rb [hunk 5]
```diff
@@ -0,0 +1,84 @@
+module PrintMachine
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/1-2.rb]: Done

File: Tasks/ruby_objects/1-1.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  def show_count_of_enemies
+    "Count of enemies on level is: #{@enemies}"
+  end
+end
+
+class MarioGame
+  include PrintMachine
+
+  attr_accessor :level_background, :count_of_enemies, :speed_enemies, :level_number
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/1-1.rb]: Done

File: Tasks/ruby_objects/1-2.rb [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+  def show_count_of_enemies
+    "Count of enemies on level is: #{@enemies}"
+  end
+end
+
+class MarioGame
+  include PrintMachine
+
+  attr_accessor :level_background, :count_of_enemies, :speed_enemies, :level_number
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/1-2.rb]: Done

File: Tasks/ruby_objects/2.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  def change_speed_enemies(new_speed_enemies)
+    @speed_enemies = new_speed_enemies
+  end
+
+  def change_count_of_enemies(new_count_of_enemies)
+    @enemies = new_count_of_enemies
+  end
+
+  def self.define_splash_screen
+    a = ObjectSpace.each_object(MarioGame).to_a
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/2.rb]: Done

File: Tasks/ruby_objects/2.rb (hunk 1 above)
Comment (NoxVail) [reply Tasks/ruby_objects/2.rb]: Done

File: Tasks/ruby_objects/PrintMachine.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+module PrintMachine
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/PrintMachine.rb]: Done

File: Tasks/ruby_objects/2.rb [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+  def change_speed_enemies(new_speed_enemies)
+    @speed_enemies = new_speed_enemies
+  end
+
+  def change_count_of_enemies(new_count_of_enemies)
+    @enemies = new_count_of_enemies
+  end
+
+  def self.define_splash_screen
```
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/2.rb]: Done

File: Tasks/ruby_objects/1-1.rb (hunk 2 above)
Comment (NoxVail) [reply Tasks/ruby_objects/1-1.rb]: Done

File: Tasks/ruby_objects/2.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (NoxVail) [reply Tasks/ruby_objects/2.rb]: I removed 3 of them, leaving only one, which is being used in `self.define_splash_screen`

File: Tasks/ruby_objects/1-2_exec.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+require_relative 'print_machine'
```
Comment (Patriarh22) [inline Tasks/ruby_objects/1-2_exec.rb]: You should do
```
  require_relative 'print_machine'
```
in the file, where this module including, I mean files with classes `MyGame ` and `MarioGame`

File: Tasks/ruby_objects/1-1.rb [hunk 12]
```diff
@@ -0,0 +1,28 @@
+require_relative 'print_machine'
+
+class MarioGame
```
Comment (Patriarh22) [inline Tasks/ruby_objects/1-1.rb]: you already have this class in file `mario_game.rb`, you just need to require it here and demonstrate how this class is working.

File: Tasks/ruby_objects/2.rb [hunk 13]
```diff
@@ -0,0 +1,47 @@
+require_relative 'print_machine'
+require 'objspace'
+
+class MarioGame
```
Comment (Patriarh22) [inline Tasks/ruby_objects/2.rb]: You have this class in a separated file, so you can just require a file with the class here and extend it in the next way
```
class MarioGame

  attr_accessor :level_number

  def self.define_splash_screen
    ...
  end
end

# execution examples
```

File: Tasks/ruby_objects/1-2_exec.rb (hunk 11 above)
Comment (NoxVail) [reply Tasks/ruby_objects/1-2_exec.rb]: Done

File: Tasks/ruby_objects/1-1.rb (hunk 12 above)
Comment (NoxVail) [reply Tasks/ruby_objects/1-1.rb]: Done

File: Tasks/ruby_objects/2.rb (hunk 13 above)
Comment (NoxVail) [reply Tasks/ruby_objects/2.rb]: Done

**Pull request merged**